clippy.toml

```diff
@@ -1 +1 @@
-msrv = "1.79"
+msrv = "1.81"
```

Comment:
I had to bump Rust due to the derive_more error stuff. I think this goes against 

> Alloy will keep a rolling MSRV policy of at least two versions behind the latest stable release (so if the latest stable release is 1.58, we would support 1.56).

I think that Rust 1.81 is also needed for https://github.com/alloy-rs/alloy/pull/1552 so maybe we should wait depending on the prio level